cherrypy/lib/cptools.py

```diff
@@ -630,3 +630,21 @@ def set_response_header():
         v.sort()
         resp_h['Vary'] = ', '.join(v)
     request.hooks.attach('before_finalize', set_response_header, 95)
+
+
+def convert_params(exception, error=400):
+    """Convert request params based on function annotations, with error handling.
+
+    exception
+        Exception class to catch.
```

Comment:
Normally I'm wary of implicit error catching, but since this is really error translating, sounds ok to me.

It's `ValueError` though; that would cover the common cases: `int`, `float`, `json.loads`, etc.